Source/Workers/createVerticesFromGoogleEarthEnterpriseBuffer.js

```diff
@@ -0,0 +1,450 @@
+                    westBorder.push({
+                        index: pointOffset,
+                        cartographic: Cartographic.clone(scratchCartographic)
+                    });
+                } else if(Math.abs(longitude - geographicEast) < halfStepX) {
+                    eastBorder.push({
+                        index: pointOffset,
+                        cartographic: Cartographic.clone(scratchCartographic)
+                    });
+                } else if (Math.abs(latitude - geographicSouth) < halfStepY) {
+                    southBorder.push({
+                        index: pointOffset,
+                        cartographic: Cartographic.clone(scratchCartographic)
+                    });
+                } else if(Math.abs(latitude - geographicNorth) < halfStepY) {
+                    northBorder.push({
+                        index: pointOffset,
+                        cartographic: Cartographic.clone(scratchCartographic)
+                    });
+                }
+
+                minHeight = Math.min(height, minHeight);
+                maxHeight = Math.max(height, maxHeight);
+                heights[pointOffset] = height;
+
+                var pos = ellipsoid.cartographicToCartesian(scratchCartographic);
+                positions[pointOffset] = pos;
+
+                if (includeWebMercatorT) {
+                    webMercatorTs[pointOffset] = (WebMercatorProjection.geodeticLatitudeToMercatorAngle(latitude) - southMercatorY) * oneOverMercatorHeight;
+                }
+
+                Matrix4.multiplyByPoint(toENU, pos, scratchCartesian);
+
+                Cartesian3.minimumByComponent(scratchCartesian, minimum, minimum);
+                Cartesian3.maximumByComponent(scratchCartesian, maximum, maximum);
+
+                var u = (longitude - geographicWest) / (geographicEast - geographicWest);
+                u = CesiumMath.clamp(u, 0.0, 1.0);
+                var v = (latitude - geographicSouth) / (geographicNorth - geographicSouth);
+                v = CesiumMath.clamp(v, 0.0, 1.0);
+
+                uvs[pointOffset] = new Cartesian2(u, v);
+                ++pointOffset;
+            }
+
+            var facesElementCount = numFaces * 3;
+            for (i = 0; i < facesElementCount; ++i, ++indicesOffset) {
+                indices[indicesOffset] = indicesMapping[dv.getUint16(offset, true)];
+                offset += sizeOfUint16;
+            }
+        }
+
+        positions.length = pointOffset;
+        uvs.length = pointOffset;
+        heights.length = pointOffset;
+        if (includeWebMercatorT) {
+            webMercatorTs.length = pointOffset;
+        }
+
+        var vertexCountWithoutSkirts = pointOffset;
+        var skirtIndex = indicesOffset;
+
+        // Add skirt points
+        var hMin = minHeight;
+        var lastBorderPoint;
+        function addSkirt(borderPoints, fudgeFactor, eastOrWest, cornerFudge) {
```

Comment:
We usually only nest functions inside functions only when needed.  Can this be moved to a top-level function?